hadoop-hdds/rocks-native/src/main/patches/rocks-native.patch

```diff
@@ -16,592 +16,557 @@
++    plain_table_options.encoding_type = kPlain;
++    plain_table_options.full_scan_mode = true;
++
++    options_.table_factory.reset(NewPlainTableFactory(plain_table_options));
++    if (!silent_) {
++      fprintf(stdout, "Sst file format: plain table\n");
++    }
++  } else {
++    char error_msg_buffer[80];
++    snprintf(error_msg_buffer, sizeof(error_msg_buffer) - 1,
++             "Unsupported table magic number --- %lx",
++             (long)table_magic_number);
++    return Status::InvalidArgument(error_msg_buffer);
++  }
++
++  return Status::OK();
++}
++
++Status RawSstFileReader::SetOldTableOptions() {
++  assert(rep_->table_properties_ == nullptr);
++  options_.table_factory = std::make_shared<BlockBasedTableFactory>();
++  if (!silent_) {
++    fprintf(stdout, "Sst file format: block-based(old version)\n");
++  }
++
++  return Status::OK();
 +}
 +
-+void write(const std::string& value, FILE* file) {
-+  write(value.data(), (int)value.length(), file);
++RawIterator* RawSstFileReader::newIterator(
++    bool has_from, Slice* from, bool has_to, Slice* to) {
++  InternalIterator* iter = rep_->table_reader_->NewIterator(
++      rep_->read_options_, rep_->moptions_.prefix_extractor.get(),
++      /*arena=*/nullptr, /*skip_filters=*/false,
++      TableReaderCaller::kSSTDumpTool);
++  return new RawSstFileReaderIterator(iter, has_from, from, has_to, to);
++
 +}
++}  // namespace ROCKSDB_NAMESPACE
++
++#endif  // ROCKSDB_LITE
+diff --git a/tools/raw_sst_file_reader_iterator.cc b/tools/raw_sst_file_reader_iterator.cc
+new file mode 100644
+index 000000000..39a2b84be
+--- /dev/null
++++ b/tools/raw_sst_file_reader_iterator.cc
+@@ -0,0 +1,94 @@
++//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
++//  This source code is licensed under both the GPLv2 (found in the
++//  COPYING file in the root directory) and Apache 2.0 License
++//  (found in the LICENSE.Apache file in the root directory).
++//
++#ifndef ROCKSDB_LITE
++
++#include "tools/raw_sst_file_reader_iterator.h"
++
++#include <chrono>
++#include <cinttypes>
++#include <iostream>
++#include <map>
++#include <memory>
++#include <sstream>
++#include <vector>
++
++#include "db/blob/blob_index.h"
++#include "db/memtable.h"
++#include "db/write_batch_internal.h"
++#include "options/cf_options.h"
++#include "port/port.h"
++#include "rocksdb/db.h"
++#include "rocksdb/env.h"
++#include "rocksdb/iterator.h"
++#include "rocksdb/slice_transform.h"
++#include "rocksdb/status.h"
++#include "rocksdb/table_properties.h"
++#include "rocksdb/utilities/ldb_cmd.h"
++#include "table/block_based/block.h"
++#include "table/block_based/block_based_table_builder.h"
++#include "table/block_based/block_based_table_factory.h"
++#include "table/block_based/block_builder.h"
++#include "table/format.h"
++#include "table/meta_blocks.h"
++#include "table/plain/plain_table_factory.h"
++#include "table/table_reader.h"
++#include "util/compression.h"
++#include "util/random.h"
```

Comment:
nit: Are all these imports used?

Reply:
yup

Reply:
I still have some doubts please double check all the imports are needed and not just here in other places as well.

Reply:
done